src/clr-angular/data/datagrid/_datagrid.clarity.scss

```diff
@@ -3,266 +3,271 @@
         width: 100%;
-        min-height: 100%; // Must be min to fix Safari bug: height: 100% won't fill flex parent. http://bit.ly/2qAPkha
-        overflow-x: auto;
-        overflow-y: hidden;
-
-        .datagrid-spinner {
-            flex: 0 0 100%;
-            margin-left: -100%;
-            display: flex;
-            justify-content: center;
-            align-items: center;
-            background: $clr-datagrid-loading-background;
-        }
-
-    }
-    .datagrid-scroll-wrapper {
-        display: flex;
-        flex-direction: row;
-        min-width: 100%;
     }
+
     .datagrid {
         display: flex;
-        flex-flow: column nowrap;
+        flex-wrap: wrap;
+        flex: 1 1 auto;
+        width: 100%;
+        margin-top: 0;
+        overflow-x: auto; // Enables x scrolling when columns are wider than the datagrid.
+        // Changed to scroll b/c I saw double scroll bars w/ auto in firefox.
         // IE & Firefox needs this
-        min-height: 1px;
+        min-height: 1px; // Do we still need this? It was ported over
+        // Overrides the table defaults so the new structure has correct corners
+        border-bottom-left-radius: 0;
+        border-bottom-right-radius: 0;
+
+        -ms-overflow-style: -ms-autohiding-scrollbar;
     }
+
     .datagrid-head, .datagrid-body, .datagrid-row, .datagrid-column, .datagrid-cell, .datagrid-fixed-column {
         display: block;
     }
-    .datagrid-table-wrapper {
-        flex: 1 1 auto;
-        display: flex;
-        flex-flow: column nowrap;
-        // IE & Firefox needs this
-        min-height: 1px;
-    }
-    .datagrid-body {
-        flex: 1 1 auto;
-        overflow-y: auto;
-        -ms-overflow-style: -ms-autohiding-scrollbar;
 
-        display: flex;
-        flex-direction: column;
+    .datagrid-body .datagrid-row {
+        border-top: $clr-table-borderwidth solid $clr-table-border-color;
+        flex-shrink: 0;
 
-        //TODO: Use the spinner min height mixin. The mixin needs to be fixed before it can be used.
-        min-height: 3rem;
-    }
-    .datagrid-head {
-        flex: 0 0 auto;
-    }
-    .datagrid-row-flex {
-        flex: 1 1 auto;
-        display: flex;
-        flex-flow: row nowrap;
-    }
-    .datagrid-column {
-        display: flex;
-        flex: 1 1 auto;
-        &.datagrid-fixed-width {
-            flex: 0 0 auto;
+        &:first-child {
+            border-top: 0;
         }
 
-        &--hidden {
-            display: none;
+        &:hover {
+            background-color: $clr-datagrid-row-hover;
         }
-    }
 
-    .datagrid-cell {
-        flex: 1 1 auto;
-        &.datagrid-fixed-width {
-            flex: 0 0 auto;
+        &.datagrid-selected {
+            color: $clr-black;
+            background-color: $clr-global-selection-color;
         }
 
-        &--hidden {
-            display: none;
-        }
-    }
+        .datagrid-action-overflow {
```

Comment:
Doesn't look like these styles were added in this PR. Is there a reason we rearranged the file? Ideally we should do this kind of cleanup in a separate PR because I can't figure out what was added and what was removed from the diff. Its fine for now but just a note for future PRs.

Reply:
I had to develop the style in a separate scss file and then copied/pasted it back into the original Basically its been re-written from scratch while porting over the original styles where it made sense.